jwst/refpix/reference_pixels.py

```diff
@@ -1431,17 +1431,54 @@ def do_subarray_corrections(self):
         log.warning("Refpix correction skipped for MIRI subarray")
         return
 
+    def get_first_goodgroup(self, integration):
+        """Get the first good group.  This is the first group with no groupdq
+        having the DO_NOT_USE bit set.
+
+        Parameters
+        ----------
+
+        integration : int
+            integration number
+
+        Returns
+        -------
+
+        group number : int
+            The number of the first good group
+
+        first_goodgroup : nddata, 2-dimensional
+            The first good group
+        """
+        ngroups = self.ngroups
+        for i in range(ngroups):
+            groupdq = self.input_model.groupdq[integration]
+            goodorbad = np.bitwise_and(groupdq[i], dqflags.group['DO_NOT_USE'])
+            badpixels = np.where(goodorbad == dqflags.group['DO_NOT_USE'])
+            if len(badpixels[0]) == 0:
```

Comment:
That's a good question. It appears that this is tailored specifically to the use case of MIRI having entire groups (first and/or last) flagged as bad. But couldn't (or shouldn't) it be handled in a more generic way such that any group in any pixel that's flagged as bad (where "bad" can be some predetermined set of flags) will get ignored and not used in fitting? Of course this would have to be handled a little differently than either saturation or jumps. For saturation, you just ignore the group and all following groups, so group n-1 becomes the final good group in a segment. Jumps also define the boundary between segments. But a generic "bad" group would just get left out of a segment, without defining the start or end of a segment.

Reply:
Whoops, sorry, the above comment was non-sequitur.  I was thinking that we were talking about the `ramp_fit` step. Never mind.

Reply:
One concern about required no 'do_not_use' flags in the entire group  is that it could be that there is no good group for subtraction.  In other words, at least one pixel has a 'do_not_use' flag in each group.  Then no subtraction would be done, right?  Is this a good idea?  I'm wondering if the criteria is for the group to subtract to be the one with no 'do_not_use' flags in the *reference* pixels?

Reply:
Karl we can leave the step the way it is (not merge this pr) and just subtract the first frame.
Then there is no need for all the logic on which frame to subtract. Read my comment below - it does not matter which frame we subtract just as long as it is the same one that is added back in the end. I say keep it simple rather than have a bunch of unneeded "DO_NOT_USE" logic.

Reply:
I fully agree we should keep this simple.   One niggling worry, why did we want it to be the first good group anyway?  I feel like we had some reason, but for the life of me I can't remember.  Can you? 
 Gotta get better at writing down the why as well as the what!

Reply:
Basically the other instrument team subtract a bias, MIRI does not subtract a bias. The reference pixel correction is a relative correction wrt to some reference frame. Only MIRI need to subtract off a reference frame in the reference pixel algorithm because all the other teams have subtracted the bias.  It is easiest to make the reference frame the first frame. I believe there was some worry then that if we do not like the first frame because of its reset effect that we should not use it. But that was the purpose of the study that I did - I added a large noise the first frame and it did not matter at all in the reference pixel correction because the correction is a relative correction and the slopes are the same whether the first frame has a hard noise added or not.

Reply:
Ok.  Many thanks for the explanations.  Much appreciated.